### PR TITLE
DAOS-4679 fixed static runable variable in test_runable()

### PR DESCRIPTION
**Pull request by wiliamhuang**
"runable" was defined as a static bool. Once a test case is determined as "skipped",
runable is set false. The following tests cases will be skipped unexpectedly.

### CODE REVIEW

**Review by wangdi1 — APPROVED** [review]
Thanks! Huang

**Review by bfaccini — APPROVED** [review]

File: src/tests/suite/daos_test_common.c [hunk 1]
```diff
@@ -628,8 +628,8 @@ d_rank_t ranks_to_kill[MAX_KILLS];
 bool
 test_runable(test_arg_t *arg, unsigned int required_nodes)
 {
-	int		 i;
-	static bool	 runable = true;
+	int	i;
+	bool	runable = true;
```
Comment (liuxuezhao) [inline src/tests/suite/daos_test_common.c:632]: this change it good, but a side-effect is now the bool type runable in stack, bool's size is less than MPI_INT, so below "MPI_Bcast(&runable, 1, MPI_INT, 0, MPI_COMM_WORLD);" may cause stack crash. My PR rebase over this patch got __stack_chk_fail() on test_runable().
I pushed https://github.com/daos-stack/daos/pull/7745 to fix it, just FYI.

Comment (wiliamhuang) [reply src/tests/suite/daos_test_common.c:632]: Good to know! Thank you very much for fixing it! I seldom use bool. I though it was stored as int.

Comment (liuxuezhao) [reply src/tests/suite/daos_test_common.c:632]: np, I thought bool is same as int also, just tested on boro it is 1B.

**Pull request merged**